azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedClusterCreate,
+		Read:   resourceArmManagedClusterRead,
+		Update: resourceArmManagedClusterCreate,
+		Delete: resourceArmManagedClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"linux_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key_data": {
```

Comment:
am I correct in thinking this isn't returned from the API? can this field be updated?

Reply:
`key_data` is returned and can be updated as far as i can tell